[TH2-2273] New AbstractRabbitRouter and metrics (#119)
th2-net/th2-common-j: README.md

* in order for the metrics to be exported, you also will need to create an instance of CommonFactory
* common JVM metrics will also be exported alongside common service metrics
* some metric labels are enumerations (`th2_type`: `MESSAGE_GROUP`, `EVENT`, `<customTag>`;`message_type`: `RAW_MESSAGE`, `MESSAGE`)

ABSTRACT METRICS:

> * th2_rabbitmq_message_size_publish_bytes (`th2_pin`, `th2_type`, `exchange`, `routing_key`): quantity of published message bytes to RabbitMQ. The message is meant any message transferred via RabbitMQ, it can contain th2 batch message or event or other content

Reviewer: ```suggestion
* th2_rabbitmq_message_size_publish_bytes (`th2_pin`, `th2_type`, `exchange`, `routing_key`): number of published message bytes to RabbitMQ. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content
```

> * th2_rabbitmq_message_publish_total (`th2_pin`, `th2_type`, `exchange`, `routing_key`): quantity of published messages to RabbitMQ. The message is meant any message transferred via RabbitMQ, it can contain th2 batch message or event or other content

Reviewer: ```suggestion
* th2_rabbitmq_message_publish_total (`th2_pin`, `th2_type`, `exchange`, `routing_key`): quantity of published messages to RabbitMQ. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content
```

> * th2_rabbitmq_message_size_subscribe_bytes (`th2_pin`, `th2_type`, `queue`): quantity of received bytes from RabbitMQ, includes bytes of messages dropped after filters. For dropped please see 'th2_message_dropped_subscribe_total' and 'th2_message_group_dropped_subscribe_total'. The message is meant any message transferred via RabbitMQ, it can contain th2 batch message or event or other content

Reviewer: ```suggestion
* th2_rabbitmq_message_size_subscribe_bytes (`th2_pin`, `th2_type`, `queue`): number of received bytes from RabbitMQ, it includes bytes of messages dropped after filters. For dropped messages information please refer to 'th2_message_dropped_subscribe_total' and 'th2_message_group_dropped_subscribe_total'. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content
```

> * th2_rabbitmq_message_process_duration_seconds (`th2_pin`, `th2_type`, `queue`): time of message processing during subscribe from RabbitMQ in seconds. The message is meant any message transferred via RabbitMQ, it can contain th2 batch message or event or other content

Reviewer: ```suggestion
* th2_rabbitmq_message_process_duration_seconds (`th2_pin`, `th2_type`, `queue`): time of message processing during subscription from RabbitMQ in seconds. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content
```

MESSAGES METRICS:
* th2_message_publish_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of published raw or parsed messages

> * th2_message_subscribe_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of received raw or parsed messages, includes dropped after filters. For dropped please see 'th2_message_dropped_subscribe_total'

Reviewer: ```suggestion
* th2_message_subscribe_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of received raw or parsed messages, includes dropped after filters. For dropped information please refer to 'th2_message_dropped_subscribe_total'
```

> * th2_message_subscribe_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of received raw or parsed messages, includes dropped after filters. For dropped information, please refer to 'th2_message_dropped_subscribe_total'

Reviewer 2: ```suggestion
* th2_message_subscribe_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of received raw or parsed messages, includes dropped after filters. For information about the number of dropped messages, please refer to 'th2_message_dropped_subscribe_total'
```

* th2_message_dropped_publish_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of published raw or parsed messages dropped after filters
* th2_message_dropped_subscribe_total (`th2_pin`, `session_alias`, `direction`, `message_type`): quantity of received raw or parsed messages dropped after filters
* th2_message_group_publish_total (`th2_pin`, `session_alias`, `direction`): quantity of published message groups 

> * th2_message_group_subscribe_total (`th2_pin`, `session_alias`, `direction`): quantity of received message groups, includes dropped after filters. For dropped please see 'th2_message_group_dropped_subscribe_total'

Reviewer: ```suggestion
* th2_message_group_subscribe_total (`th2_pin`, `session_alias`, `direction`): quantity of received message groups, includes dropped after filters. For dropped information please refer to 'th2_message_group_dropped_subscribe_total'
```

> * th2_message_group_subscribe_total (`th2_pin`, `session_alias`, `direction`): quantity of received message groups, includes dropped after filters. For dropped information, please refer to 'th2_message_group_dropped_subscribe_total'

Reviewer 2: ```suggestion
* th2_message_group_subscribe_total (`th2_pin`, `session_alias`, `direction`): quantity of received message groups, includes dropped after filters. For information about the number of dropped messages, please refer to 'th2_message_group_dropped_subscribe_total'
```

* some metric labels are enumerations (`th2_type`: `MESSAGE_GROUP`, `EVENT`, `<customTag>`;`message_type`: `RAW_MESSAGE`, `MESSAGE`)

ABSTRACT METRICS:

> * th2_rabbitmq_message_size_publish_bytes (`th2_pin`, `th2_type`, `exchange`, `routing_key`): number of published message bytes to RabbitMQ. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content

Reviewer 2: `it may contain th2 batch message or event or other content`
I'm not sure that using may modal verb is correct in this sentence. We described examples of content types instead of permission to use some of them. 
@jean-michel-perli-sonsin @andreydrobynin what do you think about this option
`The message is meant for any data transferred via RabbitMQ, for example, th2 batch message or event or custom content`

Reviewer: Hi @Nikita-Smirnov-Exactpro, thanks for the clarification. The option that you are proposing looks better and more clearer, so I agree with it.

* th2_rabbitmq_message_publish_total (`th2_pin`, `th2_type`, `exchange`, `routing_key`): quantity of published messages to RabbitMQ. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content

> * th2_rabbitmq_message_size_subscribe_bytes (`th2_pin`, `th2_type`, `queue`): number of bytes received from RabbitMQ, it includes bytes of messages dropped after filters. For dropped message information, please refer to 'th2_message_dropped_subscribe_total' and 'th2_message_group_dropped_subscribe_total'. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content

Reviewer 2: ```suggestion
* th2_rabbitmq_message_size_subscribe_bytes (`th2_pin`, `th2_type`, `queue`): number of bytes received from RabbitMQ, it includes bytes of messages dropped after filters. For information about the number of dropped message, please refer to 'th2_message_dropped_subscribe_total' and 'th2_message_group_dropped_subscribe_total'. The message is meant for any message transferred via RabbitMQ, it may contain th2 batch message or event or other content
```
@jean-michel-perli-sonsin @andreydrobynin what do you think about this option
